### ISSUE TITLE
Is itertools.cycle(loader_u) okay? It won't shuffle the data

### ISSUE THREAD

**Issue by preachermaan** [ildoonet/unsupervised-data-augmentation#3]
Hey, thanks for the repo, I'm just in the process of trying to make it work with tabular data. However, as the title says, using this approach won't shuffle the data between epochs, suggest something like the following:

iter_u = iter(loader_u) # outside of the training loop

            try:
               unlabel1, unlabel2 = next(iter_u)
            except StopIteration:
                iter_u = iter(loader_u)
                unlabel1, unlabel2 = next(iter_u) 

Or a slightly more hacky way would be to just catch any exception and then you don't need to initialize outside of the training loop:

            try:
               unlabel1, unlabel2 = next(iter_u) # dies on first pass because iter_u does not exist
            except Exception:
                iter_u = iter(loader_u)
                unlabel1, unlabel2 = next(iter_u)

**Comment by ildoonet** [comment]
Good point. I will conduct a new experiment with the fix.

**Comment by ildoonet** [comment]
@safekidda Thanks, this get me almost 1% improvements.

**Comment by preachermaan** [comment]
Nice one :)
